README.md

```diff
@@ -34,16 +32,51 @@ for future unbundling or removal wherever possible.
 
 ## Building
 
-Our Makefile should handle the build smoothly on all supported platforms. It's
-just a simple Makefile, written by hand for GNU Make, and the most complicated
-parts of its internal machinery have to do with the varied build systems of the
-bundled libraries.
-
-Useful targets are the default `all`, `clean`, and `distclean`. The last may not
-produce an entirely clean distribution directory, and runs a bundled library's
-configure script; `git clean` may be a better option.
-
-The `vere` binary is produced in `bin/urbit`.
+Urbit uses Meson build system.
+
+Some libraries which are not found in major distributions:
+- ed25519
+- http-parser legacy version 0.1.0
+- murmur3
+- softfloat3
+- urbit-scrypt
+- commonmark legacy version 0.12.0
+
+are included as git submodules. To build urbit from source, perform the following steps:
+
+## MacOS specifics
+On macos, you need to make sure `pkg-config` uses the correct homebrew path.
+ The `export PKG_CONFIG_PATH=/usr/local/opt/openssl/lib/pkgconfig/:$PKG_CONFIG_PATH`
+ should setup the `pkg-config` path correctly, solving errors with homebrew package discovery (notably with `openssl` paths).
```

Comment:
??? isn't the whole point of build systems not having to manually set environment variables

Reply:
I agree that this step is really horrible. @mikolajpp @eglaysher @ohAitch, is there any way we can automate this?

Reply:
This is because the homebrew OpenSSL formula is "keg-only": https://docs.brew.sh/How-to-build-software-outside-Homebrew-with-Homebrew-keg-only-dependencies

Homebrew has done this because Apple still shipping old libraries: https://langui.sh/2015/07/24/osx-clang-include-lib-search-paths/

Further discussion here: https://github.com/Homebrew/brew/pull/597

It looks like the options are manually linking, hardcoding paths, or setting env vars like PKG_CONFIG_PATH.

Reply:
I am fine with any of those, though presumably hardcoding the path into `PKG_CONFIG_PATH` is cleanest; this belongs *inside* the build script though, and not as a plaintext Step.

Reply:
It looks like the separation between meson and ninja makes that difficult:

- https://github.com/mesonbuild/meson/issues/541
- https://github.com/ninja-build/ninja/issues/1002#issuecomment-221169296

Reply:
Agree with @ohAitch that this should not be a user step.

Furthermore, is it possible to configure a kind of `./install` script to perform all of the build steps in one go? I.e., synchronously:

```
$ git submodule init
$ git submodule update
$ meson ./build
$ ninja -C ./build
$ sudo install -m 0755 ./build/urbit /usr/local/bin
$ urbit
```

Reply:
Like, a [Scripts to Rule Them All](https://github.com/github/scripts-to-rule-them-all) sort of thing? I'd be in favour, though that shouldn't block this PR.

Reply:
We have tried to workaround it in a proper way, but openssl on el Capitan is impossible to get detected right out of the box: we can detect the old version only.

We can sure ship with an install script which in particular takes care of this on macOs

Reply:
I have added 2 preliminary scripts: bootstrap and build which at least eliminate the need for casual user to issue any meson or build commands, and also take care of setting PKG_CONFIG_PATH to workaround MacOS